tests/event.test.ts

```diff
@@ -0,0 +1,225 @@
+
+    const pastEventsResponse = await eventController.getPastEvents({}, user);
+    const futureEventsResponse = await eventController.getFutureEvents({}, user);
+    const allEventsResponse = await eventController.getAllEvents({}, user);
+
+    expect(pastEventsResponse.events).toEqual(
+      expect.arrayContaining([pastEvent.getPublicEvent()]),
+    );
+    expect(futureEventsResponse.events).toEqual(
+      expect.arrayContaining([
+        ongoingEvent.getPublicEvent(),
+        futureEvent.getPublicEvent(),
+      ]),
+    );
+    expect(allEventsResponse.events).toEqual(
+      expect.arrayContaining([
+        pastEvent.getPublicEvent(),
+        ongoingEvent.getPublicEvent(),
+        futureEvent.getPublicEvent(),
+      ]),
+    );
+  });
+
+  test('events can be created with unused attendance codes', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event1] = EventFactory.with({ attendanceCode: 'code' });
+    const [event2] = EventFactory.with({ attendanceCode: 'different-code' });
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createEvents([event1])
+      .createUsers([admin])
+      .write();
+
+    const createEventResponse = await ControllerFactory.event(conn).createEvent({ event: event2 }, admin);
+
+    expect(createEventResponse.error).toBeNull();
+  });
+
+  test('events cannot be created with duplicate attendance codes', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event1] = EventFactory.with({ attendanceCode: 'code' });
+    const [event2] = EventFactory.with({ attendanceCode: 'code' });
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createEvents([event1])
+      .createUsers([admin])
+      .write();
+
+    const eventController = ControllerFactory.event(conn);
+    const errorMessage = 'Attendance code has already been used';
+
+    await expect(
+      eventController.createEvent({ event: event2 }, admin),
+    ).rejects.toThrow(errorMessage);
+
+    const eventsResponse = await eventController.getAllEvents({}, admin);
+
+    expect(eventsResponse.events).toHaveLength(1);
+  });
+});
+
+describe('event covers', () => {
+  const storage = new StorageUtils();
+
+  afterEach(async () => {
+    await storage.deleteAllFilesInFolder(Config.file.EVENT_COVER_UPLOAD_PATH);
+  });
+
+  test('properly updates cover photo in database and on S3', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event] = EventFactory.create(1);
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const cover = FileFactory.image(Config.file.MAX_EVENT_COVER_FILE_SIZE / 2);
+
+    await new PortalState()
+      .createUsers([admin])
+      .createEvents([event])
+      .write();
+
+    const eventCoverResponse = await ControllerFactory.event(conn).updateEventCover(cover, event.uuid, admin);
```

Comment:
I'm not too sure what we should pass in to the `StorageService` constructor, since I believe most of our config (such as bucket to be written to) is unchangeable. The `S3_BUCKET, S3_REGION`, and `S3_CREDENTIALS` are all constant through the entire app, and there's separate buckets for local (`acmucsd-local`, for what is used for locally running the test suite and in CircleCI jobs), testing (`acmucsd-membership-portal`, for testing.api.acmucsd.com), and prod (`acmucsd`, for what is used in production), so tests don't actually get written to our prod bucket.